pkg/sensors/tracing/tracepoint_test.go

```diff
@@ -824,11 +824,11 @@ spec:
         values:
         - ` + pidStr + `
       matchArgs:
-      - index: 0
+      - index: 4
         operator: "InMap"
         values:
         - "list:test"
-      - index: 1
+      - index: 5
```

Comment:
This change seems to indicate that the semantics of the `tracepoints:` changes which breaks existing policies.

Reply:
This doesn't actually break existing policies, but the way it was originally written was wrong IMO. In other tracepoints (not raw_syscalls) the arg index in the selectors matches the arg index in the args section (and there is code here: https://github.com/cilium/tetragon/blob/main/pkg/sensors/tracing/generictracepoint.go#L535 that rewrites the selector arg index (e.g. 5) to the list index of the arg that it applies to (e.g. 1). But it appears that you can either do that, or can use a 0-based index into the list of args in the args section directly (which is what the code is doing), adn this isn't documented AFAIK.

This change in the test makes no functional change to the code – so what worked before still works  – but it exercises the referenced code that rewrites the selector index number.

IMO there might be a bug here. If an args section references args with indices 1 and 2 (so ignoring the first argument to the function entirely), and the match arg selector specifies index 1, which argument does it apply to? The code will rewrite it to index 0 having matched it against the arg with index 1 (the first in the 0-indexed list), but the implication of the test (before the change) was that an arg selector with index 1 is specifying the second arg (0-indexed list of args), and therefore the arg with index 2.

I haven't addressed this directly in this PR because I think it needs discussion and a PR of its own. I did, however, modify the test so that it matches what the code expects. The test only works unmodified because an arg with index 1 doesn't exist in the args section (otherwise the index would have been rewritten).

Reply:
I will however break this change out into its own commit as I break up the other changes.